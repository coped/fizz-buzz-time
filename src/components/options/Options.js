import React from "react";
import "./Options.scss";

export default function Options(props) {
  const { toggleShowTimer, handleChange, fizz, buzz, count } = props;
  const isDisabled = count > 0 ? true : false;
  return (
    <div id="Options">
      <p className="description">
        Please enter a fizz and buzz time in seconds.{" "}
        <strong>Values should be 2 to 10, inclusive.</strong>
      </p>
      <div className="fizz-buzz-fields">
        <label id="fizz-label" htmlFor="fizz" className="label">
          Fizz:
        </label>
        <input
          id="fizz"
          className="input"
          name="fizz"
          type="text"
          value={fizz}
          onChange={handleChange}
          disabled={isDisabled}
        />
        <label id="buzz-label" htmlFor="buzz" className="label">
          Buzz:
        </label>
        <input
          id="buzz"
          className="input"
          name="buzz"
          type="text"
          value={buzz}
          onChange={handleChange}
          disabled={isDisabled}
        />
      </div>
      <button
        id="to-timer"
        onClick={toggleShowTimer}
        type="button"
        className="button"
      >
        Go to timer >
      </button>
    </div>
  );
}
