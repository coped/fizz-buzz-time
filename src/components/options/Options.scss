#Options {
  display: flex;
  height: 100%;
  width: 100%;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  .button {
    margin-top: 50px;
  }

  .fizz-buzz-fields {
    margin-top: 40px;
  }

  .label {
    font-size: 18px;
    margin-right: 5px;
  }

  .input {
    width: 100px;
    padding: 8px 16px;
    border: 1px solid #707070;
    background-color: #f5f5f5;
    border-radius: 5px;
    font-size: 16px;
    margin-right: 40px;
  }
}
