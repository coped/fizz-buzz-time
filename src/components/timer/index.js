import Timer from "components/timer/Timer"

export { Timer }