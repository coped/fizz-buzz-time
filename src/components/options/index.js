import Options from "components/options/Options";

export { Options };
