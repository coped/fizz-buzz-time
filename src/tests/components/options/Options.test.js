import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Options } from "components/options";

Enzyme.configure({ adapter: new Adapter() });

describe("Options", () => {
  const selectors = {
    fizz: "#fizz",
    buzz: "#buzz",
    fizzLabel: "#fizz-label",
    buzzLabel: "#buzz-label",
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Options />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders a fizz label", () => {
    const wrapper = shallow(<Options />);
    expect(wrapper.find(selectors.fizzLabel).length).toEqual(1);
  });

  it("renders a fizz input field", () => {
    const wrapper = shallow(<Options />);
    expect(wrapper.find(selectors.fizz).length).toEqual(1);
  });

  it("renders a buzz label", () => {
    const wrapper = shallow(<Options />);
    expect(wrapper.find(selectors.buzzLabel).length).toEqual(1);
  });

  it("renders a buzz input field", () => {
    const wrapper = shallow(<Options />);
    expect(wrapper.find(selectors.buzz).length).toEqual(1);
  });

  // it("has a valid snapshot", () => {
  //   const component = renderer.create(<Options />);
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
