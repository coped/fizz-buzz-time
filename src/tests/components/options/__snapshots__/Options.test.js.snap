// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options has a valid snapshot 1`] = `
<div
  id="Options"
>
  <p
    className="description"
  >
    Please enter a fizz and buzz time in seconds.
     
    <strong>
      Values should be 2 to 10, inclusive.
    </strong>
  </p>
  <div
    className="fizz-buzz-fields"
  >
    <label
      className="label"
      htmlFor="fizz"
      id="fizz-label"
    >
      Fizz:
    </label>
    <input
      className="input"
      disabled={false}
      id="fizz"
      name="fizz"
      type="text"
    />
    <label
      className="label"
      htmlFor="buzz"
      id="buzz-label"
    >
      Buzz:
    </label>
    <input
      className="input"
      disabled={false}
      id="buzz"
      name="buzz"
      type="text"
    />
  </div>
  <button
    className="button"
    id="to-timer"
    type="button"
  >
    Go to timer &gt;
  </button>
</div>
`;
